src/basic/fileio.c

```diff
@@ -1076,35 +1076,66 @@ int fdopen_independent(int fd, const char *mode, FILE **ret) {
         return 0;
 }
 
-static int search_and_fopen_internal(
+static int search_and_open_internal(
                 const char *path,
-                const char *mode,
+                int mode,            /* if ret_fd is NULL this is an F_OK/X_OK/R_OK/W_OK mode for access(), otherwise an open mode for open() */
                 const char *root,
                 char **search,
-                FILE **ret,
+                int *ret_fd,
                 char **ret_path) {
 
+        int r;
+
         assert(path);
-        assert(mode);
-        assert(ret);
+
+        if (path_is_absolute(path)) {
+                _cleanup_close_ int fd = -EBADF;
+                bool success;
+
+                if (ret_fd) {
+                        fd = open(path, mode, 0777);
```

Comment:
Should this even support `O_CREAT`? Maybe disallow that and remove the explicit permissions?

Reply:
i only added the permissions because one of the analyzers complained. will add comment